CONTRIBUTING.rst

```diff
@@ -203,23 +205,57 @@ formats, this could become more complicated.
 Github work flow
 ----------------
 
-1. Before diving into technicalities: if you intend to make major changes,
-   beyond fixing bugs and small functionality improvements, please open a Github
-   issue first, so we can discuss before coding. Please explain what you intend
-   to accomplish and why. That often saves a lot of time and trouble in the long
-   run.
+Before diving into technicalities: if you intend to make major changes, beyond
+fixing bugs and small functionality improvements, please open a Github issue
+first, so we can discuss before coding. Please explain what you intend to
+accomplish and why. That often saves a lot of time and trouble in the long run.
+
+Use the issue to plan your changes. Try to solve only one problem at a time,
+instead of fixing several issues and adding different features in a single shot.
+Small changes are easier to handle, also for the reviewer in the last step
+below.
+
+Mention in the corresponding issue when you are working on it. "Claim" the issue
+to avoid duplicate efforts.
+
+1. Check your GitHub settings and your local git configuration:
+
+   - If you don't have an SSH key pair yet, create one with the following
+     terminal command:
+
+     .. code-block:: bash
+
+        ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
+
+     A suitable name for this key would be ``id_rsa_github``.
+     An empty pass phrase is convenient and should be fine.
+     This will generate a private and a public key in ``${HOME}/.ssh``.
+
+   - Upload your *public* SSH key to `<https://github.com/settings/keys>`_.
+     This is a single long line in ``id_rsa_github.pub``, which you can
+     copy-paste into the browser.
+
+   - Configure SSH to use this key pair for authentication when pushing
+     branches to Github. Add the following to your ``.ssh/config`` file:
+
+     .. code-block::
+
+       Host github.ugent.be
```

Comment:
I think this should be "Host github.com", no?

Reply:
oops

Reply:
@FarnazH Thanks for catching this one! I've also add the Hostname line. It may be redundnant in shome SSH versions, but it makes things a little clearer.